### DISCUSSION TITLE
리액트 쿼리 stale time에 대해

### DISCUSSION THREAD

**Discussion by chsua** [woowacourse-teams/2023-votogether#505] [질문]
캐시에 대해 고민을 하며 찾아보던 중 한 [블로그](https://ttaerrim.tistory.com/53)를 발견하였습니다.
해당 블로그에 따르면 

staleTime과 cacheTime이 있고,
기본값은 staleTime = 0, cacheTime = 5min이라고 합니다.

때문에 api가 요청되면 staleTime에 의해 무조건 옛날 data라고 판단하고 새로 통신을 한 후 새 데이터가 변화되었다면
해당 데이터를 다시 캐싱한다고 이해했습니다.

하지만 제 기억으로는 저희는 쿼리키를 삭제해서 다시 패치를 해오는 것으로 알고 있습니다.
위 방식을 사용하는 이유는 새로고침을 하지 않고도 api통신을 통해 새로운 데이터를 보여주기 위함인지 궁금합니다. 

여담이지만 마감된 게시물의 정보는 수정될 일이 없으니 게시글 상세/투표 통계 api는 staleTime과 cacheTime를 길게 해도 좋을 것 같습니다.

**Comment by Gilpop8663** [comment]
쿼리키를 초기화 하는 로직

1. 카테고리 즐겨찾기 시 카테고리 목록을 초기화 함 - 카테고리는 캐싱, 스테일 타임이 1시간이기에 초기화를 해서 새로 불러와줘야 함
2. 댓글 생성, 삭제, 수정 시 댓글 목록을 초기화 함 - 댓글을 불러오고 5분간은 캐싱이 되어 있기 때문에 새롭게 패치해오지 않습니다. 같은 페이지에서 댓글을  추가, 삭제, 수정했을 때 즉각적으로 사용자에게 반영이 되어야 하기 때문에 캐시를 초기화하고 있습니다
3. 게시글 추가, 투표하기, 게시글 삭제 시  유저 정보를 초기화 함 - 왼쪽 유저 정보 UI에 게시글 수 , 투표 수가 사용자가 동작을 할 때마다 반영되어야 하기 때문입니다

위와 같이 쿼리 키가 초기값인 경우 같은 페이지에서 변경이 일어나야 할 때 새로운 데이터를 불러오기 위해 쿼리키를 삭제하였습니다

마감된 게시글을 불러와서 onSuccess에서 마감 여부를 통해 staletime, cachetime을 늘리는 방법이 있겠네요 좋은데요 👍👍👍
